From 3722aac31424591347e355fc61aacd30b7d1bd55 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Sun, 25 Sep 2022 22:08:35 +0300
Subject: [PATCH] [SW-0] fuse: use validation pass flag

This patch adds changes to use validation flag to retrieve spi
offset from SPI manifest. Master and Slave die offset will be
stored in persistent memory which will stored across reset
and used based on validation flag.

Change-Id: I06c38a2465625a7d8e614e796aa2b1f84928ecce
---

diff --git a/arch/gaudi3/arc/spi_flash.c b/arch/gaudi3/arc/spi_flash.c
index 86864b7..69b74ec 100644
--- a/arch/gaudi3/arc/spi_flash.c
+++ b/arch/gaudi3/arc/spi_flash.c
@@ -46,13 +46,15 @@
 	if (arcboot_rom_global.sku)
 		return;
 
-	/* Slave spi offset is stored in img_desc->spi_offset[1] */
-	idx = boot_desc_get_img_desc(HL_FW_BOOT_PTR_SPI, HL_FW_IMG_FAKE_EFUSE,
-					&desc, FLOPS_UPD_PENDING);
-	if (idx == HL_FW_IMG_NOT_FOUND)
-		return;
+	if (!arcboot_rom_global.validation_pass) {
+		/* Slave spi offset is stored in img_desc->spi_offset[1] */
+		idx = boot_desc_get_img_desc(HL_FW_BOOT_PTR_SPI, HL_FW_IMG_FAKE_EFUSE,
+						&desc, FLOPS_UPD_PENDING);
+		if (idx == HL_FW_IMG_NOT_FOUND)
+			return;
 
-	arcboot_rom_global.d1_fuse_spi_offset = desc.spi_offset;
+		arcboot_rom_global.d1_fuse_spi_offset = desc.spi_offset;
+	}
 
 	/* read value from bank0 */
 	offset = (EFUSE_SYSCFG_OFFSET * sizeof(uint32_t));
@@ -88,12 +90,14 @@
 		return;
 	}
 
-	idx = boot_desc_get_img_desc(HL_FW_BOOT_PTR_SPI, HL_FW_IMG_FAKE_EFUSE,
-					&desc, FLOPS_UPD_NOOP);
-	if (idx == HL_FW_IMG_NOT_FOUND)
-		return;
+	if (!arcboot_rom_global.validation_pass) {
+		idx = boot_desc_get_img_desc(HL_FW_BOOT_PTR_SPI, HL_FW_IMG_FAKE_EFUSE,
+						&desc, FLOPS_UPD_NOOP);
+		if (idx == HL_FW_IMG_NOT_FOUND)
+			return;
 
-	arcboot_rom_global.d0_fuse_spi_offset = desc.spi_offset;
+		arcboot_rom_global.d0_fuse_spi_offset = desc.spi_offset;
+	}
 	arcboot_rom_global.fuse_spi = true;
 
 	if (fuse_read(EFUSE_SYSCFG_OFFSET, &val) == -1)
